src/js/media-controller.js

```diff
@@ -301,6 +324,28 @@ class MediaController extends MediaContainer {
       },
     };
 
+    if (this.airplayUnavailable !== AvailabilityStates.UNSUPPORTED) {
+      const airplaySupporHandler = (event) => {
+        // NOTE: since we invoke all these event handlers without arguments whenever a media is attached,
+        // need to account for the possibility that event is undefined (CJP).
+        if (event?.availability === 'available') {
+          this._airplayUnavailable = undefined;
+        } else if (event?.availability === 'not-available') {
+          this._airplayUnavailable = AvailabilityStates.UNAVAILABLE;
+        }
+        this.propagateMediaState(
+          MediaUIAttributes.MEDIA_AIRPLAY_UNAVAILABLE,
+          this.airplayUnavailable
+        );
+      };
+      // NOTE: only adding this if airplay is supported, in part to avoid unnecessary battery consumption per
+      // Apple docs recommendations (See: https://developer.apple.com/documentation/webkitjs/adding_an_airplay_button_to_your_safari_media_controls)
```

Comment:
Do we think there's actually something under the hood that makes this necessary or is the doc writer being overly helpful in how to write javascript?

Reply:
No idea... I can speculate that having registered event listeners could cause some kind of polling of the network "under the hood", which could be a larger battery drain than a "normal" event listener. Doesn't seem too wild.